applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -170,49 +170,8 @@ void SmallStrainUPwDiffOrderElement::Initialize(const ProcessInfo& rCurrentProce
 
     UPwBaseElement::Initialize(rCurrentProcessInfo);
 
-    const auto& r_geometry        = GetGeometry();
-    const auto  number_of_U_nodes = r_geometry.PointsNumber();
-    const auto  dimension         = r_geometry.WorkingSpaceDimension();
-
-    switch (number_of_U_nodes) {
-    case 6: // 2D T6P3
-        mpPressureGeometry = make_shared<Triangle2D3<Node>>(r_geometry(0), r_geometry(1), r_geometry(2));
-        break;
-    case 8: // 2D Q8P4
-        mpPressureGeometry = make_shared<Quadrilateral2D4<Node>>(r_geometry(0), r_geometry(1),
-                                                                 r_geometry(2), r_geometry(3));
-        break;
-    case 9: // 2D Q9P4
-        mpPressureGeometry = make_shared<Quadrilateral2D4<Node>>(r_geometry(0), r_geometry(1),
-                                                                 r_geometry(2), r_geometry(3));
-        break;
-    case 10: // 3D T10P4  //2D T10P6
-        if (dimension == 3)
-            mpPressureGeometry = make_shared<Tetrahedra3D4<Node>>(r_geometry(0), r_geometry(1),
-                                                                  r_geometry(2), r_geometry(3));
-        else if (dimension == 2)
-            mpPressureGeometry = make_shared<Triangle2D6<Node>>(
-                r_geometry(0), r_geometry(1), r_geometry(2), r_geometry(3), r_geometry(4), r_geometry(5));
-        break;
-    case 15: // 2D T15P10
-        mpPressureGeometry = make_shared<Triangle2D10<Node>>(
-            r_geometry(0), r_geometry(1), r_geometry(2), r_geometry(3), r_geometry(4),
-            r_geometry(5), r_geometry(6), r_geometry(7), r_geometry(8), r_geometry(9));
-        break;
-    case 20: // 3D H20P8
-        mpPressureGeometry =
-            make_shared<Hexahedra3D8<Node>>(r_geometry(0), r_geometry(1), r_geometry(2), r_geometry(3),
-                                            r_geometry(4), r_geometry(5), r_geometry(6), r_geometry(7));
-        break;
-    case 27: // 3D H27P8
-        mpPressureGeometry =
-            make_shared<Hexahedra3D8<Node>>(r_geometry(0), r_geometry(1), r_geometry(2), r_geometry(3),
-                                            r_geometry(4), r_geometry(5), r_geometry(6), r_geometry(7));
-        break;
-    default:
-        KRATOS_ERROR << "Unexpected geometry type for different order "
-                        "interpolation element"
-                     << this->Id() << std::endl;
+    if (mpPressureGeometry == nullptr) {
+        this->SetUpPressureGeometryPointer();
```

Comment:
Since we always define the pressure geometry in the base now, we would prefer a KRATOS_DEBUG_ERROR if we still end up with a nullptr
```suggestion
    KRATOS_DEBUG_ERROR_IF_NOT(mpPressureGeometry) << "Pressure Geometry is not defined\n";
```

Reply:
Makes very much sense to me. I just wanted to keep previous behavior just in case.

Reply:
Note that I moved it to the `Check` method as this is the proper place for this. Also note that this makes possible to use the base class `Initialize` implementation.

Reply:
Thanks, `Check` is indeed the appropriate place and nice we can re-use the base implementation now!